src/components/VoteAdder.jsx

```diff
@@ -1,53 +1,48 @@
 import { useEffect, useState } from "react";
-import { updateArticleVotes } from "../api";
+import { fetchArticleById, updateArticleVotes } from "../api";
 
 export default function VoteAdder({ article, setArticle }) {
-  const [hasVoted, setHasVoted] = useState(false);
+  const [isVotedAdded, setIsVoteAdded] = useState(false);
   const [error, setError] = useState(null);
+  const [articleVotes, setArticleVotes] = useState(0);
 
-  const handleClick = () => {
-    if (!hasVoted) {
-      setHasVoted(true);
-      updateArticleVotes(article.article_id, 1)
-        .then(() => {
-          const updatedArticle = { ...article, votes: article.votes + 1 };
-          setArticle(updatedArticle);
-          localStorage.setItem(
-            `article_${article.article_id}`,
-            JSON.stringify(updatedArticle)
-          );
-        })
-        .catch((err) => {
-          setHasVoted(false);
-          setError(err);
-          console.log(err);
-        });
-    }
-  };
+  const handleClick = (event) => {
+    event.currentTarget.disabled = true;
+
+    setArticle((currArticle) => {
+      return { ...currArticle, votes: currArticle.votes + 1 };
+    });
 
-  useEffect(() => {
-    const storedArticle = JSON.parse(
-      localStorage.getItem(`article_${article.article_id}`)
-    );
-    if (storedArticle) {
-      setArticle(storedArticle);
-    }
-  }, [article.article_id]);
+    updateArticleVotes(article.article_id, 1)
+      .then((response) => {
+        setArticle({ ...response, comment_count: article.comment_count });
+        setIsVoteAdded(true);
+      })
+      .catch((err) => {
+        setError(err);
+        setArticle((currArticle) => ({
+          ...currArticle,
+          votes: currArticle.votes - 1,
+        }));
+        console.log(err);
+      });
+  };
 
   if (error) {
     return <p>Error adding your vote, please try again</p>;
   }
 
   return (
-    <button className="like-button" onClick={handleClick}>
-      <img
-        src="https://img.freepik.com/free-vector/like-button-thumbs-up-cartoon-style_78370-1159.jpg"
-        className="like-image"
-      ></img>
-    </button>
+    <>
+      <button className="like-button" onClick={handleClick}>
+        <img
+          src="https://img.freepik.com/free-vector/like-button-thumbs-up-cartoon-style_78370-1159.jpg"
```

Comment:
Because this has a white background, it is not clear when it gets shaded out to indicate it's disabled. You could use an emoji 👍  or an image with a transparent background to get a better effect